software/base/src/main/java/org/apache/brooklyn/entity/software/base/AbstractSoftwareProcessDriver.java

```diff
@@ -116,11 +116,9 @@ public void start() {
                 preInstall();
             }});
 
-            if (Strings.isNonBlank(entity.getConfig(BrooklynConfigKeys.PRE_INSTALL_COMMAND))) {
-                DynamicTasks.queue("pre-install-command", new Runnable() { public void run() {
-                    runPreInstallCommand(entity.getConfig(BrooklynConfigKeys.PRE_INSTALL_COMMAND));
-                }});
-            };
+            DynamicTasks.queue("pre-install-command", new Runnable() { public void run() {
```

Comment:
Seems wrong to have "pre-install-command" as a special case. There is still a check that the command is non-blank for `POST_INSTALL_COMMAND`, `PRE_LAUNCH_COMMAND` and `POST_LAUNCH_COMMAND`. We should either include the if check for all of them or for none of them.